fs2-google-pubsub/src/main/scala/com/permutive/pubsub/consumer/decoder/MessageDecoder.scala

```diff
@@ -17,18 +17,29 @@
 package com.permutive.pubsub.consumer.decoder
 
 import cats.Functor
+import cats.syntax.all._
 
 trait MessageDecoder[A] {
+
   def decode(message: Array[Byte]): Either[Throwable, A]
+
+  def map[B](f: A => B): MessageDecoder[B] = MessageDecoder.functor.map(this)(f)
+
+  def emap[B](f: A => Either[Throwable, B]): MessageDecoder[B] = this.decode(_).flatMap(f)
+
 }
 
 object MessageDecoder {
+
   def apply[A: MessageDecoder]: MessageDecoder[A] = implicitly
 
+  val string: MessageDecoder[String] = new String(_).asRight
```

Comment:
Would it be safer to wrap this in a Either.catchNonFatal()?

Reply:
In addition, should we be explicit with the encoding?

Reply:
Great suggestion! Done https://github.com/permutive-engineering/fs2-google-pubsub/pull/436/commits/94f075466b9dd750a761e34370a75b3b0c21a575